src/components/Schedule/Schedule.tsx

```diff
@@ -0,0 +1,270 @@
+  addService: (newInfo: NewServiceData) => void;
+  removeSchedule: (info: DeleteScheduleProps) => void;
+}
+
+// Task List
+// 1. When the container has more than 3 tabs, when you select the 4th, how to handle data from data handler
+// 2. Need to update changedTask and the buttons to reflect only on the schedule selected
+// 3. Cell data poorly describes the different kinds of cells that exist. The data structure needs a revamp, and so does the Tablecell/Datacell
+// 4. Low priority but finding scheduleId and order is a pain the way I have it currently implemented because allScheduleData exists in tabs, but singular schedule data exist in schedules. Need to consolidate some of the logic together
+
+export const Schedule = ({
+  tabs,
+  state,
+  addSchedule,
+  removeSchedule,
+  addService,
+}: ScheduleProps) => {
+  const { isLoading, error, data, isSuccess } = state;
+  const [tab, setTab] = useState(0);
+  const [isScheduleModified, setIsScheduleModified] = useState<boolean>(false);
+  const [isNewScheduleOpen, setIsNewScheduleOpen] = useState<boolean>(false);
+  const [isNewServiceOpen, setIsNewServiceOpen] = useState<boolean>(false);
+  //   const [alert, setAlert] = useState<AlertInterface>();
+
+  const changedTasks = useRef<any>({});
+  const outerRef = useRef(null);
+
+  // Schedule
+  function onChangeTabs(value: number) {
+    if (value === tabs.length) return;
+    // fetchSchedule(value);
+    // I think I need to wait for data to refetch, can't test until we have more schedules to work with
+    setTab(value);
+  }
+  // Save Data
+  function onTaskModified(taskId: number, newAssignee: number, isChanged: boolean) {
+    if (isChanged) {
+      const updatedChangedTasks = { ...changedTasks.current, [taskId]: newAssignee };
+      changedTasks.current = updatedChangedTasks;
+    } else if (changedTasks.current[taskId]) delete changedTasks.current[taskId];
+
+    setIsScheduleModified(Object.keys(changedTasks.current).length > 0);
+  }
+  function onSaveScheduleChanges() {
+    setIsScheduleModified(false);
+  }
+
+  // Service Body
+  function deleteRow(rowIndex: number) {}
+  function insertRow(rowIndex: number) {}
+
+  useEffect(() => {
+    // with react-query, there is no awaiting for a response to finish. so the results need to be handled by the mutations.isSuccess/isError/etc methods in data handler
+    if (isSuccess === 'NewScheduleForm') {
+      setIsNewScheduleOpen(false);
+      setTab(tabs.length);
+    }
+    if (isSuccess === 'NewServiceForm') setIsNewServiceOpen(false);
+    if (isSuccess === 'DeleteSchedule') {
+      // setIsNewScheduleOpen(false);
+      setTab(0);
+    }
+    if (isSuccess === 'DeleteService') {
+      // setIsNewScheduleOpen(false);
+    }
+  }, [isSuccess]);
+
+  const teammates = (roleId) =>
+    data.users.filter((user: any) => user.teams.some((role: any) => role.id === roleId));
+
+  return (
+    <>
+      {data?.schedules && data?.users && !isLoading ? (
```

Comment:
for easier reading, i'd "flip" the ternary condition ([de morgan](https://en.wikipedia.org/wiki/De_Morgan%27s_laws#cite_ref-3:~:text=the%20negation%20of%20a%20conjunction%20is%20the%20disjunction%20of%20the%20negations)) so we can move the shorter alternative output to be directly under the condition, like this:
```suggestion
      {!data?.schedules || !data?.users || isLoading ? (
          <CircularProgress />
      ) : (
          /* the longer, main schedule tabs stuff */
      )}
```
Instead of having the `<CircularProgress />` suddenly at the end.

Reply:
Oo, what's the reasoning behind this? Ted and I were talking about how while developing, it's a tad bit annoying when there's much fluff nearer to the top.

Reply:
we can leave it as-is. i'd say it's fine either way, or depends. maybe i'm overapplying a rule, but it's like moving short guard clauses to the top of functions. 

Instead of wondering what the else condition is, readers know immediately - this makes sense when the alternate condition is really short. So instead of this:

```js
if (condition) {
  // long
  // long
  // long
  // long
  // long
  // ...
} else { // huh? scroll back up to find the matching brace
  // short
}
```

and replace that with this:

```js
if (!condition) {
  // short
} else {
  // long
  // long
  // long
  // long
  // long
}
```

but thinking about it from the other point of view:

```js
if (condition) {
  // main
  // happy
  // path
  // goes
  // here
} else {
  // edge case you don't usually care about
}
```

Reply:
hmm this makes sense to me. @CedThen what do you think? If you're ok with it, I'll change it, and we'll follow by these comments, if not also ok